tools/config_gen.py

```diff
@@ -188,7 +189,7 @@ def _print_options(opt_group, options):
         print(("# %s" % opt.help).strip())
 
         if isinstance(opt, cfg.StrOpt) and opt.type.choices:
-            if isinstance(opt.type.choices, list):
+            if isinstance(opt.type.choices, OrderedDict):
```

Comment:
@nzlosh Is this really an `OrderedDict` in the newer versions of oslo.config?

Reply:
I believe so.  To confirm I preformed the following:

1. Reverted to list:
```
(virtualenv) [root@rocky9 st2]# git diff tools/config_gen.py                                      
diff --git a/tools/config_gen.py b/tools/config_gen.py                                                                                                                                             
index f139474c1..d1f82b8f1 100755
--- a/tools/config_gen.py                                                                                                                                                                          
+++ b/tools/config_gen.py                
@@ -189,7 +189,7 @@ def _print_options(opt_group, options):                           
         print(("# %s" % opt.help).strip())
                                                                                                 
         if isinstance(opt, cfg.StrOpt) and opt.type.choices:
-            if isinstance(opt.type.choices, OrderedDict):                                                                                                                                         
+            if isinstance(opt.type.choices, list):
                 valid_values = ", ".join([str(x) for x in opt.type.choices])
             else:
                 valid_values = opt.type.choices 
```
2. Re-create conf sampe
```
(virtualenv) [root@rocky9 st2]# ./tools/config_gen.py > conf/st2.conf.sample 
```
3. Show difference
```
(virtualenv) [root@rocky9 st2]# git diff conf/st2.conf.sample 
diff --git a/conf/st2.conf.sample b/conf/st2.conf.sample
index 82da2ae2e..a97a5a35f 100644
--- a/conf/st2.conf.sample
+++ b/conf/st2.conf.sample
@@ -39,7 +39,7 @@ workflows_pool_size = 40
 # List of origins allowed for api, auth and stream
 allow_origin = http://127.0.0.1:3000 # comma separated list allowed here.
 # SameSite attribute value for the auth-token cookie we set on successful authentication from st2web. If you don't have a specific reason (e.g. supporting old browsers) we recommend you set this
 value to strict. Setting it to "unset" will default to the behavior in previous releases and not set this SameSite header value.
-# Valid values: strict, lax, none, unset
+# Valid values: OrderedDict([('strict', None), ('lax', None), ('none', None), ('unset', None)])
 auth_cookie_same_site = lax
 # True if secure flag should be set for "auth-token" cookie which is set on successful authentication via st2web. You should only set this to False if you have a good reason to not run and acces
s StackStorm behind https proxy.
 auth_cookie_secure = True
```

Reply:
Looks like they're using it as an ordered set, ignoring the dict values. That's weird, but ok. 🤷‍♂️